### ISSUE TITLE
After installation of Node v0.12.4 Error:undefined symbol: _ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE

### ISSUE THREAD

**Issue by mfurqan777** [joyent/node#25483]
Hi
OS: 64-Ubuntu
I have install latest node v0.12.4 but
When starting the node following error occurred:

/home/node/out/Release/node: symbol lookup error: /home/nodejs/node-mobile/node_modules/ws/build/Release/bufferutil.node: undefined symbol: _ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE
[2015-06-05T12:28:22.503Z] (sys) Starting
/home/node/out/Release/node: symbol lookup error: /home/nodejs/node-mobile/node_modules/ws/build/Release/bufferutil.node: undefined symbol: _ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE
[2015-06-05T12:47:47.341Z] (sys) Starting
/home/node/out/Release/node: symbol lookup error: /home/nodejs/node-mobile/node_modules/ws/build/Release/bufferutil.node: undefined symbol: _ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE

Any suggestions or guide

**Comment by fivdi** [comment]
This appears to be related to binary addon `bufferutil` which `ws` depends on. Binary addons are built against the version of node that's installed when the addon is installed. After upgrading to a new version of node, v0.12.4 in this case, they probably need to be rebuilt.

**Comment by misterdjules** [comment]
TL;DR: between node v0.10.x and node v0.12.x, `v8::Object::Set(v8::Handle<v8::Value>, v8::Handle<v8::Value>, v8::PropertyAttribute)` changed to `v8::Object::Set(v8::Handle<v8::Value>, v8::Handle<v8::Value>)` (there are possibly other variants available). So it seems that `bufferutil.node` was linked against node v0.10.x. Now that you're using node v012.x you will need to rebuild `bufferutil.node` by running `npm rebuild`.

Longer story:

`__ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE` is the mangled version of `v8::Object::Set(v8::Handle<v8::Value>, v8::Handle<v8::Value>, v8::PropertyAttribute)`:
```
$ echo "__ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE" | c++filt -s auto
v8::Object::Set(v8::Handle<v8::Value>, v8::Handle<v8::Value>, v8::PropertyAttribute)
$
```

This symbol is present in node v0.10.x:

```
$ node --version
v0.10.38
$ nm `which node` | grep _ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE
0000000100127fb0 T __ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE
$
```
But not in node v0.12.x:
```
$ ./out/Release/node --version
v0.12.0
$ nm ./out/Release/node | grep _ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_NS_17PropertyAttributeE 
$
```
Instead in node v0.12.x a different signature for the same API, `v8::Object::Set(v8::Handle<v8::Value>, v8::Handle<v8::Value>)`, is available:
```
$ nm ./out/Release/node | grep _ZN2v86Object3SetENS_6HandleINS_5ValueEEES3                        
00000001001266e0 T __ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_
$ echo "__ZN2v86Object3SetENS_6HandleINS_5ValueEEES3_" | c++filt -s auto
v8::Object::Set(v8::Handle<v8::Value>, v8::Handle<v8::Value>)
$
```

Closing as it doesn't seem to be an issue with node, but please feel free to comment if you have any question.

Thanks!
